python/lsst/ts/planning/tool/zephyr_interface.py

```diff
@@ -156,12 +166,39 @@ async def get_steps_in_test_case(self, test_case_key):
         --------
         * https://support.smartbear.com/zephyr-scale-cloud/api-docs/\
                 #tag/Test-Cases/operation/getTestCaseTestSteps
+        * https://support.smartbear.com/zephyr-scale-cloud/api-docs/\
+                #tag/Test-Executions/operation/getTestExecutionTestSteps
         """
-        endpoint = f"testcases/{test_case_key}/teststeps"
-        self.log.info(f"Querying steps in test case {test_case_key}")
-        return await self.get(endpoint)
+        if re.search(r"(.+-E[0-9]+)", test_key):
+            param_key = "testexecutions"
+            self.log.info(f"Querying steps in test execution {test_key}")
+        elif re.search(r"(.+-T[0-9]+)", test_key):
+            param_key = "testcases"
+            self.log.info(f"Querying steps in test case {test_key}")
+        else:
+            raise ValueError("Invalid test key")
+
+        endpoint = f"{param_key}/{test_key}/teststeps"
+        payload = await self.get(endpoint)
+
+        if complement and param_key == "testcases":
```

Comment:
It seems the "call to test" logic is only for queries using a Test Case Key. I understand you are querying the same endpoint so that's why you kept Test Steps of Test Cases and Test Executions together, but I think it would be good to separate them in different methods such as: `get_steps_test_case` & `get_steps_test_execution`. That way you don't have to be using extra arguments to handle the logic and you just make "call to test" on the `get_steps_test_case` method.
Also all the logic from line 184 to 197 I would move that to a different function, something like `get_steps_call_to_test(json_steps: list)`